spec/system/user_page/user_preferences_profile_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+describe "User preferences | Profile", type: :system do
+  fab!(:user) { Fabricate(:user, active: true) }
+  let(:user_preferences_profile_page) { PageObjects::Pages::UserPreferencesProfile.new }
+  let(:user_preferences_page) { PageObjects::Pages::UserPreferences.new }
+
+  before { sign_in(user) }
+
+  describe "enforcing required fields" do
+    before do
+      UserRequiredFieldsVersion.create!
+      UserField.create!(
+        field_type: "text",
+        name: "Favourite Pokemon",
+        description: "Hint: It's Mudkip.",
+        requirement: :for_all_users,
+        editable: true,
+      )
+    end
+
+    it "allows user to fill up required fields" do
+      # Redirects to the profile page to fill up missing fields.
+      #
+      visit("/")
+
+      expect(page).to have_current_path("/u/bruce0/preferences/profile")
+
+      # Shows a notice that the user needs to fill up more fields.
```

Comment:
Like I said in chat too I really don't think the comments are necessary and we should split the redirect test into another test. Thanks for doing the I18n stuff though!